builder/builder.go

```diff
@@ -112,13 +110,13 @@ type Backend interface {
 	// TODO: use digest reference instead of name
 
 	// GetImage looks up a Docker image referenced by `name`.
-	GetImage(name string) (*image.Image, error)
+	GetImage(name string) (Image, error)
 	// Pull tells Docker to pull image referenced by `name`.
-	Pull(name string) (*image.Image, error)
-	// ContainerWsAttachWithLogs attaches to container.
-	ContainerWsAttachWithLogs(name string, cfg *daemon.ContainerWsAttachWithLogsConfig) error
```

Comment:
nit picking about having a struct for the new `ContainerAttach` to be somehow consistent with other methods - `ContainerRm` has a struct for just 3 arguments for instance